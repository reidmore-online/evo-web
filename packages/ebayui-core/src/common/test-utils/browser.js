import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
