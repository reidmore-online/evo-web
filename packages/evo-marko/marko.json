{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
