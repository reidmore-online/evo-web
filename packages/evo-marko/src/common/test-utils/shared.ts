export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
