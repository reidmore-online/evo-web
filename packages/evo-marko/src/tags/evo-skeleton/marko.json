{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
