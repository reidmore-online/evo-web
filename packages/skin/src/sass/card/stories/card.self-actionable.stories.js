export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
