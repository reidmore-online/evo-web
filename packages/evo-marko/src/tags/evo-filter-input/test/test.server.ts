import { describe, it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../filter-input.stories";

const { Default, Controls } = composeStories(stories);

describe("evo-filter-input", () => {
  it("renders default input", async () => {
    await snapshotHTML(Default);
  });

  it("renders small", async () => {
    await snapshotHTML(Default, { size: "small" });
  });

  it("renders large", async () => {
    await snapshotHTML(Default, { size: "large" });
  });

  it("renders with aria-controls", async () => {
    await snapshotHTML(Controls);
  });

  it("renders without clear button and no placeholder", async () => {
    await snapshotHTML(Default, { a11yClearButton: "", placeholder: "" });
  });

  it("renders with custom placeholder and aria label", async () => {
    await snapshotHTML(Default, {
      placeholder: "Custom placeholder",
      "aria-label": "Custom aria label",
      a11yClearButton: "Clear the text",
    });
  });
});
