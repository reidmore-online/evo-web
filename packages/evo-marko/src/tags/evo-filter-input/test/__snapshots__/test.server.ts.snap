// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evo-filter-input > renders default input 1`] = `
<DocumentFragment>
  <span
    class="filter-input"
  >
    <div
      class="textbox textbox--fluid"
      placeholder=""
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 16 16"
        >
          <path
            d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
          />
        </symbol>
        <use
          href="#icon-search-16"
        />
      </svg>
      <input
        aria-describedby="sM_2 sM_3"
        aria-label="Filter input"
        class="textbox__control"
        id="GENERATED-1"
        type="search"
      />
      <button
        aria-label="Clear filter input"
        class="icon-btn icon-btn--transparent icon-btn--small"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-2"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm3.71 10.29a1 1 0 1 1-1.41 1.41L8 9.41l-2.29 2.3A1 1 0 0 1 4.3 10.3L6.59 8l-2.3-2.29a1.004 1.004 0 0 1 1.42-1.42L8 6.59l2.29-2.29a1 1 0 0 1 1.41 1.41L9.41 8l2.3 2.29Z"
            />
          </symbol>
          <use
            href="#icon-clear-16"
          />
        </svg>
      </button>
    </div>
  </span>
</DocumentFragment>
`;

exports[`evo-filter-input > renders large 1`] = `
<DocumentFragment>
  <span
    class="filter-input filter-input--large"
  >
    <div
      class="textbox textbox--fluid"
      placeholder=""
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 16 16"
        >
          <path
            d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
          />
        </symbol>
        <use
          href="#icon-search-16"
        />
      </svg>
      <input
        aria-describedby="sM_2 sM_3"
        aria-label="Filter input"
        class="textbox__control"
        id="GENERATED-1"
        type="search"
      />
      <button
        aria-label="Clear filter input"
        class="icon-btn icon-btn--transparent icon-btn--small"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-2"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm3.71 10.29a1 1 0 1 1-1.41 1.41L8 9.41l-2.29 2.3A1 1 0 0 1 4.3 10.3L6.59 8l-2.3-2.29a1.004 1.004 0 0 1 1.42-1.42L8 6.59l2.29-2.29a1 1 0 0 1 1.41 1.41L9.41 8l2.3 2.29Z"
            />
          </symbol>
          <use
            href="#icon-clear-16"
          />
        </svg>
      </button>
    </div>
  </span>
</DocumentFragment>
`;

exports[`evo-filter-input > renders small 1`] = `
<DocumentFragment>
  <span
    class="filter-input filter-input--small"
  >
    <div
      class="textbox textbox--fluid"
      placeholder=""
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 16 16"
        >
          <path
            d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
          />
        </symbol>
        <use
          href="#icon-search-16"
        />
      </svg>
      <input
        aria-describedby="sM_2 sM_3"
        aria-label="Filter input"
        class="textbox__control"
        id="GENERATED-1"
        type="search"
      />
      <button
        aria-label="Clear filter input"
        class="icon-btn icon-btn--transparent icon-btn--small"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-2"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm3.71 10.29a1 1 0 1 1-1.41 1.41L8 9.41l-2.29 2.3A1 1 0 0 1 4.3 10.3L6.59 8l-2.3-2.29a1.004 1.004 0 0 1 1.42-1.42L8 6.59l2.29-2.29a1 1 0 0 1 1.41 1.41L9.41 8l2.3 2.29Z"
            />
          </symbol>
          <use
            href="#icon-clear-16"
          />
        </svg>
      </button>
    </div>
  </span>
</DocumentFragment>
`;

exports[`evo-filter-input > renders with aria-controls 1`] = `
<DocumentFragment>
  <div>
    <span
      class="filter-input"
    >
      <div
        class="textbox textbox--fluid"
        placeholder=""
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-0"
            viewBox="0 0 16 16"
          >
            <path
              d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
            />
          </symbol>
          <use
            href="#icon-search-16"
          />
        </svg>
        <input
          aria-controls="filter-input-controls"
          aria-describedby="sM_2 sM_3"
          aria-label="Filter input"
          class="textbox__control"
          id="GENERATED-1"
          type="search"
          value=""
        />
        <button
          aria-label="Clear filter input"
          class="icon-btn icon-btn--transparent icon-btn--small"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--16"
          >
            <symbol
              id="GENERATED-2"
              viewBox="0 0 16 16"
            >
              <path
                d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm3.71 10.29a1 1 0 1 1-1.41 1.41L8 9.41l-2.29 2.3A1 1 0 0 1 4.3 10.3L6.59 8l-2.3-2.29a1.004 1.004 0 0 1 1.42-1.42L8 6.59l2.29-2.29a1 1 0 0 1 1.41 1.41L9.41 8l2.3 2.29Z"
              />
            </symbol>
            <use
              href="#icon-clear-16"
            />
          </svg>
        </button>
      </div>
    </span>
    <div
      id="filter-input-controls"
    >
      Current value: "‍"
    </div>
  </div>
</DocumentFragment>
`;

exports[`evo-filter-input > renders with custom placeholder and aria label 1`] = `
<DocumentFragment>
  <span
    class="filter-input"
  >
    <div
      class="textbox textbox--fluid"
      placeholder=""
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 16 16"
        >
          <path
            d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
          />
        </symbol>
        <use
          href="#icon-search-16"
        />
      </svg>
      <input
        aria-describedby="sM_2 sM_3"
        aria-label="Custom aria label"
        class="textbox__control"
        id="GENERATED-1"
        type="search"
      />
      <button
        aria-label="Clear the text"
        class="icon-btn icon-btn--transparent icon-btn--small"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
        >
          <symbol
            id="GENERATED-2"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0Zm3.71 10.29a1 1 0 1 1-1.41 1.41L8 9.41l-2.29 2.3A1 1 0 0 1 4.3 10.3L6.59 8l-2.3-2.29a1.004 1.004 0 0 1 1.42-1.42L8 6.59l2.29-2.29a1 1 0 0 1 1.41 1.41L9.41 8l2.3 2.29Z"
            />
          </symbol>
          <use
            href="#icon-clear-16"
          />
        </svg>
      </button>
    </div>
  </span>
</DocumentFragment>
`;

exports[`evo-filter-input > renders without clear button and no placeholder 1`] = `
<DocumentFragment>
  <span
    class="filter-input"
  >
    <div
      class="textbox textbox--fluid"
      placeholder=""
    >
      <svg
        aria-hidden="true"
        class="icon icon--16"
      >
        <symbol
          id="GENERATED-0"
          viewBox="0 0 16 16"
        >
          <path
            d="M3 6.5a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0Zm11.76 6.85-.021-.01-3.71-3.681-.025-.008A5.465 5.465 0 0 0 12 6.5 5.5 5.5 0 1 0 6.5 12a5.47 5.47 0 0 0 3.118-.972l3.732 3.732a1 1 0 0 0 1.41-1.41Z"
          />
        </symbol>
        <use
          href="#icon-search-16"
        />
      </svg>
      <input
        aria-describedby="sM_2 sM_3"
        aria-label="Filter input"
        class="textbox__control"
        id="GENERATED-1"
        type="search"
      />
    </div>
  </span>
</DocumentFragment>
`;
